src/ig/IGExporter.ts

```diff
@@ -200,6 +181,65 @@ export class IGExporter {
     }
   }
 
+  /**
+   * Fixes a dependsOn entry by specifying its uri (if not yet specified) and generating an id (if
+   * not yet specified). Will also ensure that required properties (uri/version) are available.
+   * If it cannot ensure a valid dependsOn entry, it will return undefined.
+   * @param dependency {ImplementationGuideDependsOn} - the dependency to fix
+   * @param igs {List<Object>} - the IGs to search when finding the dependency URI
+   * @returns the fixed dependency or null if it can't be fixed
+   */
+  private fixDependsOn(dependency: ImplementationGuideDependsOn, igs: any[]) {
+    // Clone it so we don't mutate the original
+    const dependsOn: ImplementationGuideDependsOn = cloneDeep(dependency);
+
+    if (dependsOn.version == null) {
+      // No need for the detailed log message since we already logged one in the package loader.
+      logger.error(
+        `Failed to add ${dependency.packageId} to ImplementationGuide instance because no ` +
+          'version was specified in your config.yaml.'
+      );
+      return;
+    }
+
+    if (dependsOn.uri == null) {
+      // Need to find the URI from the IG in the FHIR cache
+      const dependencyIG = igs.find(
+        ig =>
+          ig.packageId === dependsOn.packageId &&
+          (ig.version === dependsOn.version ||
+            'current' === dependsOn.version ||
+            'dev' === dependsOn.version)
+      );
+      dependsOn.uri = dependencyIG?.url;
+
+      if (dependsOn.uri == null) {
+        logger.error(
+          `Failed to add ${dependsOn.packageId}:${dependsOn.version} to ` +
+            'ImplementationGuide instance because SUSHI could not find the IG URL in the ' +
+            'dependency IG. To specify the IG URL in your config.yaml, use the dependency ' +
+            'details format:\n\n' +
+            'dependencies:\n' +
+            `  ${dependsOn.packageId}:\n` +
+            '    uri: http://my-fhir-ig.org/ImplementationGuide/123\n' +
+            `    version: ${dependsOn.version}`
+        );
+        return;
+      }
+    }
+
+    if (dependsOn.id == null) {
+      // packageId should be "a..z, A..Z, 0..9, and _ and it must start with a..z | A..Z" per
+      // https://chat.fhir.org/#narrow/stream/215610-shorthand/topic/SUSHI.200.2E12.2E7/near/199193333
+      // depId should be [A-Za-z0-9\-\.]{1,64}, so we replace . and - with _ and prepend "id_" if it does not start w/ a-z|A-Z
+      dependsOn.id = /[A-Za-z]/.test(dependsOn.packageId[0])
+        ? dependsOn.packageId.replace(/\.|-/g, '_')
+        : 'id_' + dependsOn.packageId.replace(/\.|-/g, '_');
+    }
```

Comment:
Should we even still do this now that they can specify an `id`? This made sense when it was impossible for the user. But I'm not sure, now that a user can specify it, it seems kinda weird to force this hacked together `id` upon them, especially when the IG Publisher still runs perfectly fine without it, and most users probably aren't using the features that go along with this `id` in the IG Publisher.

Reply:
Maybe at least in the case that they are using the more complex syntax to specify `uri` we don't specify `id`? But I'm still not sure it's a good idea at all anymore.

Reply:
If the user doesn't use the `id`, then there is no harm in us "hacking together" one.  The alternative is that IG Publisher yells at them (with a warning) forcing them to declare an `id` despite the fact that they never use it.

In other words, I suspect that people who use the `id` will probably want to set their own anyway, and people who don't will appreciate that we did something to make the IG Publisher shut up.

Does that make you feel any better about it?

Reply:
Yeah that does make me feel better. I wasn't thinking of it from the "please be quiet IG Publisher" perspective.